### PR TITLE
Wallet: Add functions for wallet using the endpoints provided by Stoa

### PR DESCRIPTION
**Pull request by MichaelKim20**
Stoa has endpoints made for wallets. I added this to SDK for easy use.

### CODE REVIEW

File: src/modules/net/BOAClient.ts [hunk 1]
```diff
@@ -346,6 +347,127 @@ export class BOAClient
                 });
         });
     }
+
+    /**
+     * Requests a history of transactions.
```
Comment (TrustHenry) [inline src/modules/net/BOAClient.ts:352]:
```suggestion
     * Request a history of transactions.
```

File: src/modules/net/BOAClient.ts [hunk 2]
```diff
@@ -346,6 +347,127 @@ export class BOAClient
...
+                .filename(address.toString())
+                .addSearch("pageSize", page_size.toString())
+                .addSearch("page", page.toString())
+                .addSearch("type", type.join(","));
+
+            if ((begin !== undefined) && (end !== undefined))
+            {
+                url.addSearch("beginDate", begin.toString());
+                url.addSearch("endDate", end.toString());
+            }
```
Comment (TrustHenry) [inline src/modules/net/BOAClient.ts:383]: Is there any case where there's only `beginDate`?

File: src/modules/net/BOAClient.ts [hunk 3]
```diff
@@ -346,6 +347,127 @@ export class BOAClient
...
+                })
+                .catch((reason: any) =>
+                {
+                    reject(handleNetworkError(reason));
+                });
+        });
+    }
+
+    /**
+     * Requests pending transactions.
```
Comment (TrustHenry) [inline src/modules/net/BOAClient.ts:441]:
```suggestion
     * Request pending transactions.
```

File: src/modules/net/BOAClient.ts [hunk 4]
```diff
@@ -346,6 +347,127 @@ export class BOAClient
...
+     * @param address The input address of the pending transaction
+     */
+    public getWalletTransactionsPending (address: PublicKey): Promise<Array<IPendingTxs>>
+    {
+        return new Promise<Array<IPendingTxs>>((resolve, reject) =>
+        {
+            let url = uri(this.server_url)
+                .directory("/wallet/transactions/pending")
+                .filename(address.toString());
```
Comment (TrustHenry) [inline src/modules/net/BOAClient.ts:450]: Required value input check is required.

File: src/modules/net/response/Types.ts [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    License:
+        MIT License. See LICENSE for details.
+
+ *******************************************************************************/
+
+/**
+ * The interface of the transactions history element
+ */
+export interface ITxHistoryElement
```
Comment (TrustHenry) [inline src/modules/net/response/Types.ts:17]: There is a type interface duplicate code in SDK and STOA.
We should consider using this in Stoa later.

Comment (MichaelKim20) [reply src/modules/net/response/Types.ts:17]: Yes, then I'll externalize it.

File: src/modules/net/BOAClient.ts (hunk 2 above)
Comment (MichaelKim20) [reply src/modules/net/BOAClient.ts:383]: We'd better leave everything to Stoa.

File: src/modules/net/BOAClient.ts (hunk 4 above)
Comment (MichaelKim20) [reply src/modules/net/BOAClient.ts:450]: Isn't PublicKey already verified?

**Review by TrustHenry — APPROVED** [review]

**Pull request merged**